mypy/types.py

```diff
@@ -144,6 +144,97 @@ def deserialize(cls, data: JsonDict) -> 'Type':
         raise NotImplementedError('Cannot deserialize {} instance'.format(cls.__name__))
 
 
+class TypeAliasType(Type):
+    """
+    A type alias to another type.
+
+    To support recursive type aliases we don't immediately expand a type alias
```

Comment:
Add a prominent note that this is not being used yet, and the implementation is still incomplete?